lib/location/Kconfig

```diff
@@ -0,0 +1,176 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig LOCATION
+	bool "Location"
+	depends on NRF_MODEM_LIB
+	help
+	  TODO
+
+if LOCATION
+
+config LOCATION_METHOD_GNSS
+	bool "Allow GNSS to be used for obtaining the location"
+	default y
+
+config LOCATION_METHOD_CELLULAR
+	bool "Allow cellular positioning to be used for obtaining the location"
+	select MULTICELL_LOCATION
+	default y
+
+config LOCATION_METHOD_WLAN
+	bool "Allow WLAN positioning to be used for obtaining the location"
+	depends on WIFI
+	select SREST_CLIENT_LIB
```

Comment:
We need to not eliminate nRF Cloud use over MQTT.  Selecting the SREST_CLIENT_LIB implies that WiFi positioning can never work on an nRF Cloud MQTT device.

Reply:
Thanks @plskeggs for the comments. Maybe here is some misunderstanding but I don't understand how the use of SREST_CLIENT is eliminating something. It is just a generic REST client (i.e. common stuff that are always needed at the top of http lib, nothing that is wlan / rest service related), and actually a base for that was taken from nrf cloud rest lib. And eventually, also nrf cloud maybe should start to use that and get rid of the corresponding logic in there.

WLAN specific positioning stuff are currently all (nRF Cloud/Skyhook/Here support) in Location lib. nRF Cloud lib maybe should have that nRF part at some point if that is wanted. I see this pretty much similar than is multicell lib.

Reply:
I was concerned that, by selecting a REST library, this means that the MQTT transport for nRF Cloud will not be supported here (in the future, when nRF Cloud supports WLAN), but also generally by the whole library.  Perhaps I am extrapolating too broadly.

While the option of using REST from a device to nRF Cloud is great to have, nRF Cloud will continue to support MQTT.  It doesn't make sense to add all these great and useful location features to this library but not support MQTT.

Reply:
We are working on MQTT support at the moment. Some version should be ready within 1 or 2 weeks.